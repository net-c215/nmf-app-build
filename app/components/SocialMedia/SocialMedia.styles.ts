import { StyleSheet } from "react-native";
import layout from "../../style/layout";

export default StyleSheet.create({
  container: {
    ...layout.containerWithPadding,
    flexDirection: "row",
    justifyContent: "space-between",
    paddingVertical: 18
  }
});
