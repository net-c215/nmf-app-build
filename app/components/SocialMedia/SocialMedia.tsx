import React from "react";
import { View, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Linking } from "expo";

import styles from "./SocialMedia.styles";

const socialMedia = [
  {
    iconName: "instagram",
    url: "https://www.instagram.com/nmf_earth/",
    color: "#c32aa3"
  },
  {
    iconName: "slack",
    url:
      "https://join.slack.com/t/not-my-fault/shared_invite/enQtODMzMzcyNjExMzc3LTVlNDQwMTA0OGUxMmJmNTE4MTA4MTgwZGZmNTBiZmRhNTcxY2FlN2MzZGQyMmRjZDVjN2Q2MjFlNzE4ZGI5ZTI",
    color: "#4a154b"
  },
  {
    iconName: "github-circle",
    url: "https://github.com/NotMyFaultEarth",
    color: "#333333"
  },
  {
    iconName: "twitter",
    url: "https://twitter.com/nmf_earth",
    color: "#2b90d9"
  },
  {
    iconName: "mastodon",
    url: "https://mastodon.social/@nmf_earth",
    color: "#2b90d9"
  },
  {
    iconName: "medium",
    url: "https://medium.com/@nmf_earth",
    color: "#02b875"
  }
];

export default () => (
  <View style={styles.container}>
    {socialMedia.map((item, index) => (
      <TouchableOpacity key={index} onPress={() => Linking.openURL(item.url)}>
        <MaterialCommunityIcons
          name={item.iconName}
          size={32}
          color={item.color}
        />
      </TouchableOpacity>
    ))}
  </View>
);
