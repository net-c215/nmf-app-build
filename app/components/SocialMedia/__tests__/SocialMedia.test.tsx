import React from "react";
import { create } from "react-test-renderer";
import * as Linking from "expo-linking";

import SocialMedia from "../";

jest.unmock("../");

beforeEach(() => {
  jest.spyOn(Linking, "openURL").mockImplementation((url: string) => {
    if (url) {
      return Promise.resolve(true);
    }
    Promise.reject(false);
  });
});

test("renders correctly SocialMedia", () => {
  const tree = create(<SocialMedia />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should open link", () => {
  const root = create(<SocialMedia />).root;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  const icons = root.findAllByType("TouchableOpacity");
  icons.map((item) => item.props.onPress());
  expect(Linking.openURL).toHaveBeenCalledTimes(icons.length);
});
