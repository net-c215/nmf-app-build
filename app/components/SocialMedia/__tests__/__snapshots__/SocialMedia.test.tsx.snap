// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly SocialMedia 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 14,
      "paddingVertical": 18,
    }
  }
>
  <TouchableOpacity
    onPress={[Function]}
  >
    <MaterialCommunityIcons
      color="#c32aa3"
      name="instagram"
      size={32}
    />
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <MaterialCommunityIcons
      color="#4a154b"
      name="slack"
      size={32}
    />
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <MaterialCommunityIcons
      color="#333333"
      name="github-circle"
      size={32}
    />
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <MaterialCommunityIcons
      color="#2b90d9"
      name="twitter"
      size={32}
    />
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <MaterialCommunityIcons
      color="#2b90d9"
      name="mastodon"
      size={32}
    />
  </TouchableOpacity>
  <TouchableOpacity
    onPress={[Function]}
  >
    <MaterialCommunityIcons
      color="#02b875"
      name="medium"
      size={32}
    />
  </TouchableOpacity>
</View>
`;
